From 43317bc28d093d1a5fa47f9b205b96df77405651 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Tue, 15 Oct 2013 10:30:51 +0800
Subject: [PATCH 475/486] ENGR00215174 fix camera cts failure in one cycle
 run.

replace semphore wait with timeout wait to
fix camera cts failure in one cycle run.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx6/libcamera2/StreamAdapter.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/mx6/libcamera2/StreamAdapter.cpp b/mx6/libcamera2/StreamAdapter.cpp
index dd40c22..58cd23b 100755
--- a/mx6/libcamera2/StreamAdapter.cpp
+++ b/mx6/libcamera2/StreamAdapter.cpp
@@ -241,7 +241,14 @@ void StreamAdapter::handleCameraFrame(CameraFrame *frame)
 
 void StreamAdapter::applyRequest()
 {
-    sem_wait(&mRespondSem);
+    struct timespec ts;
+
+    clock_gettime(CLOCK_REALTIME, &ts);
+    long msecs = 500000000 + ts.tv_nsec;//500ms
+    long secs = msecs / 1000000000;//get seconds.
+    ts.tv_sec += secs;
+    ts.tv_nsec = msecs % 1000000000;//get milliseconds.
+    sem_timedwait(&mRespondSem, &ts);
 }
 
 void StreamAdapter::convertNV12toYV12(StreamBuffer* dst, StreamBuffer* src)
-- 
1.8.0

